classes/class-p4-search.php

```diff
@@ -265,10 +286,29 @@ protected function get_timber_posts( $paged = 1 ) : array {
 			// Use Timber's Post instead of WP_Post so that we can make use of Timber within the template.
 			if ( $posts ) {
 				foreach ( $posts as $post ) {
-					$timber_posts[] = new TimberPost( $post->ID );
+					if ( $post->post_type === P4_Post_archive::POST_TYPE ) {
+						$archive_post = new stdClass();
+						$archive_post->title = $post->post_title;
+						$archive_post->link = $post->guid;
+						$archive_post->post_type = P4_Post_archive::POST_TYPE;
+						$archive_post->post_date = $post->post_date_gmt;
+						$timber_posts[] = $archive_post;
+					} else {
+						// TODO: Not use TimberPost as that will query the data again. We just called ElasticSearch,
+						// so if we need more data, we just need to pull it from there (might require setting up more
+						// indexing.
+						$timber_post = new TimberPost( $post->ID );
```

Comment:
Notice how the `TimberPost` that is passed to the template is created with only the ID from ElasticSearch as input. This means that, even though we got most of the data from ES, this will just reload it from the db again, which kinda defeats the point of using ElasticSearch. After the change in this PR this luckily only happens for the 5 posts that are being displayed. 

But currently, in production, if you search for something that matches over 300 results, it will fetch all 300 of them again, one query at a time, even though it already has all data.